src/module.c

```diff
@@ -13512,6 +13541,278 @@ const char* RM_GetInternalSecret(RedisModuleCtx *ctx, size_t *len) {
+
+    sds config_name = sdsnew(name);
+    int ret = moduleGetNumericConfig(config_name, res);
+    sdsfree(config_name);
+    return ret ? REDISMODULE_OK : REDISMODULE_ERR;
+}
+
+/* Set the value of a config.
+ *
+ * This function can be used to set the value of any config, regardless of its
+ * type. If the config is multi-argument, the value must be a space-separated
+ * string.
+ *
+ * If the value failed to be set `err` will be set to a human readable error
+ * message and REDISMODULE_ERR will be returned. */
+int RM_SetStringConfig(RedisModuleCtx *ctx, const char *name, const char *value, RedisModuleString **err) {
+    sds config_name = sdsnew(name);
+    const char *cerr = NULL;
+    int res = moduleSetStringConfig(ctx->client, config_name, value, &cerr);
+    sdsfree(config_name);
+    if (err && cerr) {
+        *err = RM_CreateString(ctx, cerr, strlen(cerr));
+    }
+    return (res == 0 ? REDISMODULE_ERR : REDISMODULE_OK);
+}
+
+/* Set the value of a bool config.
+ *
+ * See RedisModule_SetStringConfig for return value. */
+int RM_SetBoolConfig(RedisModuleCtx *ctx, const char *name, int value, RedisModuleString **err) {
+    const char *cerr = NULL;
+    sds config_name = sdsnew(name);
+    int res = moduleSetBoolConfig(ctx->client, config_name, value, &cerr);
+    sdsfree(config_name);
+    if (err && cerr) {
+        *err = RM_CreateString(ctx, cerr, strlen(cerr));
+    }
+    return (res == 0 ? REDISMODULE_ERR : REDISMODULE_OK);
+}
+
+/* Set the value of an enum config.
+ *
+ * If the config is multi-argument one the value parameter must be a
+ * space-separated string.
+ *
+ * See RedisModule_SetStringConfig for return value. */
+int RM_SetEnumConfig(RedisModuleCtx *ctx, const char *name, const char **values, int num_values, RedisModuleString **err) {
+    sds config_name = sdsnew(name);
+    const char *cerr = NULL;
+
+    sds *sds_values = zmalloc(sizeof(sds) * num_values);
+    for (int i = 0; i < num_values; i++) {
+        sds_values[i] = sdsnew(values[i]);
+    }
+
+    int res = moduleSetEnumConfig(ctx->client, config_name, sds_values, num_values, &cerr);
+
+    for (int i = 0; i < num_values; i++) {
+        sdsfree(sds_values[i]);
+    }
+    zfree(sds_values);
+    sdsfree(config_name);
+    if (err && cerr) {
+        *err = RM_CreateString(ctx, cerr, strlen(cerr));
+    }
+    return (res == 0 ? REDISMODULE_ERR : REDISMODULE_OK);
+}
+
+/* Set the value of a numeric config.
+ * If the value passed is meant to be a percentage, it should be passed as a
+ * negative value.
+ *
+ * See RedisModule_SetStringConfig for return value. */
+int RM_SetNumericConfig(RedisModuleCtx *ctx, const char *name, long long value, RedisModuleString **err) {
+    sds config_name = sdsnew(name);
+    const char *cerr = NULL;
+    int res = moduleSetNumericConfig(ctx->client, config_name, value, &cerr);
+    sdsfree(config_name);
+    if (err && cerr) {
+        *err = RM_CreateString(ctx, cerr, strlen(cerr));
+    }
```

Comment:
if there is only one line in the `if`, we always omit the `{}`.
```suggestion
    if (err && cerr)
        *err = RM_CreateString(ctx, cerr, strlen(cerr));
```